lib/lxc.sh

```diff
@@ -3,10 +3,15 @@
 # Check for LXC or Incus
 function switch_lxc_incus()
 {
-    if [[ "${YNHDEV_BACKEND:-}" == "incus" ]]; then
-        lxc=incus
+    if [[ "${YNHDEV_BACKEND:-foo}" == "foo" ]]; then
+        # Autodetect because no preference was specified
+        if command -v incus 2>&1 >/dev/null; then
+            lxc="incus"
+        else
+            lxc="lxc"
+        fi
     else
-        lxc=lxc
+        lxc="${YNHDEV_BACKEND}"
```

Comment:
Imho having this variable called `lxc` is confusing ... `YNHDEV_BACKEND` seemed like a decent name reflecting the fact that it could be either incus or lxc/lxd ... imho we should set this variable (YNHDEV_BACKEND) if it's empty, and otherwise use the value provided

Reply:
I think that's a great idea! It should be used to source a different "backend" script with functions. But for the moment `$lxc` is used in all the commands and rarely matched against so that would be a much bigger change.